lxd/container_lxc.go

```diff
@@ -5492,23 +5542,23 @@ func (c *containerLXC) templateApplyNow(trigger string) error {
 		// Read the template
 		tplString, err := ioutil.ReadFile(filepath.Join(c.TemplatesPath(), tpl.Template))
 		if err != nil {
-			return err
+			return errors.Wrap(err, "Failed to read template file")
 		}
 
 		// Restrict filesystem access to within the container's rootfs
 		tplSet := pongo2.NewSet(fmt.Sprintf("%s-%s", c.name, tpl.Template), template.ChrootLoader{Path: c.RootfsPath()})
 
 		tplRender, err := tplSet.FromString("{% autoescape off %}" + string(tplString) + "{% endautoescape %}")
 		if err != nil {
-			return err
+			return errors.Wrap(err, "Failed to render template")
 		}
 
 		// Figure out the architecture
 		arch, err := osarch.ArchitectureName(c.architecture)
 		if err != nil {
 			arch, err = osarch.ArchitectureName(c.state.OS.Architectures[0])
 			if err != nil {
-				return err
+				return errors.Wrap(err, "Failed to figure architecture")
```

Comment:
Maybe `Failed to detect architecture`?

Reply:
`Failed to detect system architecture` would make sense, yeah, not that this ever really happens :)

Reply:
Yeah this is mainly a help for development. When you are changing (aka breaking) stuff and you get errors, it's important to know what code path triggered them.